### PR TITLE
Add ability to return immediately when a lock cannot be obtained inst…

### PR DESCRIPTION
**Pull request by myheroyuki**
Addresses issue #57

### CODE REVIEW

**Comment by jugmac00** [comment]
There is a typo in `ImmediateAquireError`  - I think this should be `ImmediateAcquireError`

**Review by FFY00 — COMMENTED** [review]
Wouldn't `blocking`/`nonblocking` be a more standard name?

Eg. https://docs.python.org/3/library/threading.html#threading.Lock.acquire

**Comment by myheroyuki** [comment]
I chose return_immediately because it was suggested in the original issue, but I agree blocking/nonblocking is the more standard convention. I updated the PR to change to blocking.

**Comment by jugmac00** [comment]
There is one flake 8 issue.

File: src/filelock/_api.py [hunk 1]
```diff
@@ -116,6 +116,7 @@ def acquire(
         poll_interval: float = 0.05,
         *,
         poll_intervall: float | None = None,
+        blocking=True,
```
Comment (gaborbernat) [inline src/filelock/_api.py]: A better solution wouldn't just use the value 0 to mean non-blocking? Do we need this extra argument?

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/filelock/_api.py (hunk 1 above)
Comment (myheroyuki) [reply src/filelock/_api.py]: As discussed in the issue #57, a timeout value of 0 currently means poll-indefinitely. Adding an additional argument makes it's functionality more clear without breaking backwards compatibility.

**Comment by gaborbernat** [comment]
You'll need to fix the CI 👍

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**